hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/SystemContractAbis.java

```diff
@@ -0,0 +1,166 @@
+    CREATE_NON_FUNGIBLE_TOKEN_V3(3, "createNonFungibleToken(%s)".formatted(HEDERA_TOKEN_STRUCT_V3)),
+
+    CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V1("createNonFungibleTokenWithCustomFees(%s,%s[],%s[])"
+            .formatted(HEDERA_TOKEN_STRUCT_V1, FIXED_FEE_V1, ROYALTY_FEE_V1)),
+    CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V2("createNonFungibleTokenWithCustomFees(%s,%s[],%s[])"
+            .formatted(HEDERA_TOKEN_STRUCT_V2, FIXED_FEE_V1, ROYALTY_FEE_V1)),
+    CREATE_NON_FUNGIBLE_TOKEN_WITH_CUSTOM_FEES_V3("createNonFungibleTokenWithCustomFees(%s,%s[],%s[])"
+            .formatted(HEDERA_TOKEN_STRUCT_V3, FIXED_FEE_V2, ROYALTY_FEE_V2)),
+
+    CRYPTO_TRANSFER_V1(1, "cryptoTransfer(%s[])".formatted(TOKEN_TRANSFER_LIST_V1)),
+    CRYPTO_TRANSFER_V2(2, "cryptoTransfer(%s,%s[])".formatted(TRANSFER_LIST_V1, TOKEN_TRANSFER_LIST_V2)),
+
+    MINT_TOKEN_V1(1, "mintToken(address,uint64,bytes[])"),
+    MINT_TOKEN_V2(2, "mintToken(address,int64,bytes[])"),
+
+    UPDATE_TOKEN_EXPIRY_INFO_V1(1, "updateTokenExpiryInfo(address,%s)".formatted(EXPIRY_V1)),
+    UPDATE_TOKEN_EXPIRY_INFO_V2(2, "updateTokenExpiryInfo(address,%s)".formatted(EXPIRY_V2)),
+
+    UPDATE_TOKEN_INFO_V1(1, "updateTokenInfo(address,%s)".formatted(HEDERA_TOKEN_STRUCT_V1)),
+    UPDATE_TOKEN_INFO_V2(2, "updateTokenInfo(address,%s)".formatted(HEDERA_TOKEN_STRUCT_V2)),
+    UPDATE_TOKEN_INFO_V3(3, "updateTokenInfo(address,%s)".formatted(HEDERA_TOKEN_STRUCT_V3)),
+
+    WIPE_TOKEN_ACCOUNT_V1(1, "wipeTokenAccount(address,address,uint32)"),
+    WIPE_TOKEN_ACCOUNT_V2(2, "wipeTokenAccount(address,address,int64)");
+
+    SystemContractAbis(int version, @NonNull final String signature) {
+        this.version = version;
+        this.signature = stripBlanks(signature);
+        this.decoderSignature = stripName(addressToUint(this.signature));
+        this.decoder = TypeFactory.create(this.decoderSignature);
+        this.selector = Bytes.wrap(new Function(this.signature).selector());
+        this.methodName = nameFrom(this.signature);
+    }
+
+    SystemContractAbis(@NonNull final String signature) {
+        this(1, signature);
+    }
+
+    public final int version;
+    public final String signature;
+    public final String decoderSignature;
+    public final ABIType<Tuple> decoder;
+    public final Bytes selector;
+    public final String methodName;
+
+    @NonNull
+    public String selectorAsHex() {
+        return to8CharHexString(selector);
+    }
+
+    @NonNull
+    String nameFrom(@NonNull final String s) {
+        return s.split("[(]", 2)[0];
+    }
+
+    @NonNull
+    String addressToUint(@NonNull final String s) {
+        return s.replace("address", "bytes32");
+    }
+
+    @NonNull
+    String stripBlanks(@NonNull final String s) {
+        return s.replaceAll("\\s+", "");
+    }
+
+    @NonNull
+    String stripName(@NonNull final String s) {
+        return s.substring(s.indexOf('('));
```

Comment:
nit: a comment with an example of the string being destructed gives insight to the string manipulation work , if not an extra comment